libraries/AP_InertialSensor/AP_InertialSensor_LSM9DS1.cpp

```diff
@@ -0,0 +1,573 @@
+#   define LSM9DS1XG_FIFO_CTRL_FMODE_BYPASS       (0x0 << 5)
+#   define LSM9DS1XG_FIFO_CTRL_FMODE_FIFO         (0x1 << 5)
+#   define LSM9DS1XG_FIFO_CTRL_FMODE_STREAM       (0x3 << 5)
+#   define LSM9DS1XG_FIFO_CTRL_FMODE_B_TO_S       (0x4 << 5)
+#   define LSM9DS1XG_FIFO_CTRL_FMODE_CON          (0x5 << 5)
+#define LSM9DS1XG_FIFO_SRC                              0x2F
+#   define LSM9DS1XG_FIFO_SRC_FTH                 (0x1 << 7)
+#   define LSM9DS1XG_FIFO_SRC_OVRN                (0x1 << 6)
+#   define LSM9DS1XG_FIFO_SRC_UNREAD_SAMPLES            0x3F
+#define LSM9DS1XG_INT_GEN_CFG_G                         0x30
+#   define LSM9DS1XG_INT_GEN_CFG_G_AOI_G          (0x1 << 7)
+#   define LSM9DS1XG_INT_GEN_CFG_G_LIR_G          (0x1 << 6)
+#   define LSM9DS1XG_INT_GEN_CFG_G_ZHIE_G         (0x1 << 5)
+#   define LSM9DS1XG_INT_GEN_CFG_G_ZLIE_G         (0x1 << 4)
+#   define LSM9DS1XG_INT_GEN_CFG_G_YHIE_G         (0x1 << 3)
+#   define LSM9DS1XG_INT_GEN_CFG_G_YLIE_G         (0x1 << 2)
+#   define LSM9DS1XG_INT_GEN_CFG_G_XHIE_G         (0x1 << 1)
+#   define LSM9DS1XG_INT_GEN_CFG_G_XLIE_G         (0x1 << 0)
+#define LSM9DS1XG_INT_GEN_THS_XH_G                      0x31
+#   define LSM9DS1XG_INT_GEN_THS_XH_G_DCRM_G      (0x1 << 7)
+#define LSM9DS1XG_INT_GEN_THS_XL_G                      0x32
+#define LSM9DS1XG_INT_GEN_THS_YH_G                      0x33
+#define LSM9DS1XG_INT_GEN_THS_YL_G                      0x34
+#define LSM9DS1XG_INT_GEN_THS_ZH_G                      0x35
+#define LSM9DS1XG_INT_GEN_THS_ZL_G                      0x36
+#define LSM9DS1XG_INT_GEN_DUR_G                         0x37
+#   define LSM9DS1XG_INT_GEN_DUR_G_WAIT_G         (0x1 << 7)
+
+
+
+AP_InertialSensor_LSM9DS1::AP_InertialSensor_LSM9DS1(AP_InertialSensor &imu,
+                                                     AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev,
+                                                     int drdy_pin_num_xg,
+                                                     enum Rotation rotation)
+    : AP_InertialSensor_Backend(imu)
+    , _dev(std::move(dev))
+    , _drdy_pin_num_xg(drdy_pin_num_xg)
+    , _rotation(rotation)
+
+{
+}
+
+AP_InertialSensor_Backend *AP_InertialSensor_LSM9DS1::probe(AP_InertialSensor &_imu,
+                                                            AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev,
+                                                            enum Rotation rotation)
+{
+    if (!dev) {
+        return nullptr;
+    }
+
+    AP_InertialSensor_LSM9DS1 *sensor =
+        new AP_InertialSensor_LSM9DS1(_imu,std::move(dev),
+                                      LSM9DS1_DRY_XG_PIN,
+                                      rotation);
+    if (!sensor || !sensor->_init_sensor()) {
+        delete sensor;
+        return nullptr;
+    }
+
+    return sensor;
+}
+
+bool AP_InertialSensor_LSM9DS1::_init_sensor()
+{
+    /*
+     * Same semaphore for both since they necessarily share the same bus (with
+     * different CS)
```

Comment:
if you have 2 different CS, you should have 2 different `AP_HAL::SPIDevice`.  From your driver and datasheet it seems this comment is a leftover from copying LSM9DS0 since you have a single CS for accel/gyro and another CS for mag.